WordPress/Classes/Utility/ImmuTable.swift

```diff
@@ -0,0 +1,206 @@
+    case Nib(UINib)
+    case Class(AnyClass)
+}
+
+public protocol CellRegistrator {
+    func register(registrable: CellRegistrable, cellReuseIdentifier: String)
+}
+
+
+extension UITableView: CellRegistrator {
+    public func register(registrable: CellRegistrable, cellReuseIdentifier: String) {
+        switch registrable {
+        case .Nib(let nib):
+            registerNib(nib, forCellReuseIdentifier: cellReuseIdentifier)
+        case .Class(let cellClass):
+            registerClass(cellClass, forCellReuseIdentifier: cellReuseIdentifier)
+        }
+    }
+}
+
+public struct ImmuTable {
+    public let sections: [ImmuTableSection]
+
+    public func rowAtIndexPath(indexPath: NSIndexPath) -> ImmuTableRow {
+        return sections[indexPath.section].rows[indexPath.row]
+    }
+
+    public static func registerRows(rows: [ImmuTableRow.Type], tableView: CellRegistrator) {
+        let registrables = rows.reduce([:]) {
+            (var classes, row) -> [String: CellRegistrable] in
+
+            classes[row.reusableIdentifier] = row.registrable
+            return classes
+        }
+        for (identifier, registrable) in registrables {
+            tableView.register(registrable, cellReuseIdentifier: identifier)
+        }
+    }
+}
+
+public class ImmuTableDataSource: NSObject, UITableViewDataSource {
+    var viewModel: ImmuTable
+    var configureCell: ((UITableViewCell) -> Void)?
+
+    init(viewModel: ImmuTable) {
+        self.viewModel = viewModel
+    }
+
+    public func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return viewModel.sections.count
+    }
+
+    public func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.sections[section].rows.count
+    }
+
+    public func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let row = viewModel.rowAtIndexPath(indexPath)
+        let cell = tableView.dequeueReusableCellWithIdentifier(row.reusableIdentifier, forIndexPath: indexPath)
+
+        row.configureCell(cell)
+
+        configureCell?(cell)
+
+        return cell
+    }
+}
+
+public class ImmuTableDelegate: NSObject, UITableViewDelegate {
+    var viewModel: ImmuTable
+
+    init(viewModel: ImmuTable) {
+        self.viewModel = viewModel
+    }
+
+    public func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        let row = viewModel.rowAtIndexPath(indexPath)
+        if let action = row.action {
+            action(row)
+        }
+    }
+
+    public func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        let row = viewModel.rowAtIndexPath(indexPath)
+        if let customHeight = row.dynamicType.customHeight {
+            return CGFloat(customHeight)
+        }
+        return tableView.rowHeight
+    }
+}
+
+public struct ImmuTableViewHandler {
+    let target: UITableViewController
```

Comment:
So this would be easy to implement

https://gist.github.com/koke/b9b7f2e675b6132501bc

Another option would be to make the UITableView the target. Then instead of having to ask the target if the view is loaded, the handler could have a `isViewLoaded` property, or just check if `tableView.superview != nil`. Or maybe we just call reloadData anyway, even if that means calling it twice. To be honest, the current behaviour is tied to UITableViewController:

> When the table view is about to appear the first time it’s loaded, the table-view controller reloads the table view’s data

If we can't be sure that the target VC is a UITableViewController, we can't rely on that behaviour and we should be reloading data when the model is set.

Thoughts?